src/lib.rs

```diff
@@ -67,6 +67,7 @@
 #![deny(missing_docs)]
 #![deny(meta_variable_misuse)]
 #![no_std]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
```

Comment:
If you switch to `doc_auto_cfg`, you no longer need `doc_cfg`, and by omitting `doc_cfg` you can also guard against people adding manual annotations (if docs are tested in CI, that is)

```suggestion
#![cfg_attr(docsrs, feature(doc_auto_cfg))]
```